src/subdomains/core/buy-crypto/process/services/buy-crypto-preparation.service.ts

```diff
@@ -120,13 +126,17 @@ export class BuyCryptoPreparationService {
       // Only for bankTx/checkoutTx BuyCrypto
       try {
         const inputReferenceCurrency = await this.fiatService.getFiatByName(entity.inputReference.currency);
+        const inputCurrency = await this.fiatService.getFiatByName(entity.inputAsset);
 
-        const { feeAmount, fee, minFee } = await this.transactionHelper.getTxDetails(
-          entity.inputReferenceAmount,
-          undefined,
-          inputReferenceCurrency,
+        const inputReferencePrice = await this.priceProviderService.getPrice(inputCurrency, inputReferenceCurrency);
```

Comment:
`inputReferencePrice` is already fixed -> just use `inputAmount / inputReferenceAmount`?